armi/settings/settingsIO.py

```diff
@@ -275,6 +275,9 @@ def _readYaml(self, stream, handleInvalids=True):
         for settingName, settingVal in caseSettings.items():
             self._applySettings(settingName, settingVal)
 
+        if handleInvalids:
+            self._checkInvalidSettings()
```

Comment:
I recommend moving this to `readFromStream` and pulling out the handleInvalids arg to that level. Then you can remove the identical code from `_readFromXml` while you're at it.

Reply:
I'll try it!

Reply:
Yeah, that seems like a clean refactor. It also allows us to clean up the method declarations for both of those.

Done!